### PR TITLE
add recaptcha, workflow, and other things to make site more secure

### PR DESCRIPTION
**Pull request by JDDoesDev**
### Changes

* Adds the workflow, content moderation, recaptcha and captcha modules
* Adds a topic and training submission workflows
* Updates permissions for the above
* Ignore config for recaptcha keys
* Update honeypot settings to be less lenient

### CODE REVIEW

File: conf/drupal/config/user.role.authenticated.yml [hunk 1]
```diff
@@ -14,11 +14,11 @@ permissions:
   - 'access shortcuts'
   - 'access site-wide contact form'
   - 'access user profiles'
-  - 'create topic content'
   - 'edit own topic content'
   - 'post comments'
   - 'search content'
   - 'skip comment approval'
+  - 'use session_proposal transition create_new_draft'
```
Comment (Brian-Clement) [inline conf/drupal/config/user.role.authenticated.yml]: Should this user also have permissions for the `submitted_proposal`?  At the minute they can only create content in the "Draft" state.

File: conf/drupal/config/user.role.authenticated.yml [hunk 2]
```diff
@@ -14,11 +14,11 @@ permissions:
   - 'access shortcuts'
   - 'access site-wide contact form'
   - 'access user profiles'
-  - 'create topic content'
```
[the remark this answers is not in the document]
Comment (JDDoesDev) [reply conf/drupal/config/user.role.authenticated.yml]: Until we open up sessions I think we should keep this permission disabled for all users, and then during session submission time we should open it back up to authenticated users.  That will also help curb some of the issues with spam submissions year round.

**Comment by JDDoesDev** [comment]
Just added in a little descriptive text to hopefully make the workflow a little more clear to session submitters.

File: conf/drupal/config/user.role.authenticated.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Brian-Clement) [reply conf/drupal/config/user.role.authenticated.yml]: That works for me.

**Pull request merged**